Alliance.Client/Extensions/VOIP/Behaviors/PBVoiceChatHandlerClient.cs

```diff
@@ -1,1175 +1,1065 @@
+			{
+				if (_voiceData.Count > 0)
+				{
+					bool isMutedFromGameOrPlatform = Peer?.IsMutedFromGameOrPlatform ?? false;
+					if (_playDelayRemainingSizeInBytes > 0)
+					{
+						_playDelayRemainingSizeInBytes -= 2;
+					}
+					else
+					{
+						short item = _voiceData.Dequeue();
+						_nextPlayDelayResetTime = MissionTime.Now + MissionTime.Milliseconds(PlayDelayResetTimeInMilliseconds);
+
+						if (!isMutedFromGameOrPlatform)
+						{
+							for (int i = 0; i < _voiceData.Count; i++)
+							{
+								item = _voiceData.Dequeue();
+								_voiceToPlayInTick.Enqueue(item);
+							}
+						}
+					}
+
+					return !isMutedFromGameOrPlatform;
+				}
+
+				return false;
+			}
+
+			public void ClearVoiceData(MixingSampleProvider audioMixer)
+			{
+				_voiceData.Clear();
+				_voiceToPlayInTick.Clear();
+				_nextPlayDelayResetTime = MissionTime.Now;
+				_playDelayRemainingSizeInBytes = 0;
+
+				audioMixer.RemoveMixerInput(volumeProvider);
+			}
+
+			public Queue<short> GetVoiceToPlayForTick()
+			{
+				return _voiceToPlayInTick;
+			}
+
+			public bool HasAnyVoiceData()
+			{
+				return _voiceData.Count > 0;
+			}
+
+			public int GetVoiceData()
+			{
+				return _voiceData.Count;
+			}
+
+			public OpusDecoder GetDecoder()
+			{
+				return _decoder;
+			}
+
+			public void UpdateVoice(byte[] byteArray, float pbOptionVolume)
+			{
+				if (Peer == null && Agent == null)
+				{
+					waveProvider.ClearBuffer();
+
+					return;
+				}
+
+				Vec3 speakerPosition;
+
+				if (Peer?.ControlledAgent != null)
+				{
+					speakerPosition = Peer.ControlledAgent.Position;
+				}
+				else if (Agent != null)
+				{
+					speakerPosition = Agent.Position;
+				}
+				else if (GameNetwork.MyPeer.ControlledAgent != null)
+				{
+					speakerPosition = GameNetwork.MyPeer.ControlledAgent.Position;
+				}
+				else
+				{
+					speakerPosition = Mission.Current.GetCameraFrame().origin;
+				}
+
+				Vec3 listenerPosition = GameNetwork.MyPeer.ControlledAgent != null ? GameNetwork.MyPeer.ControlledAgent.Position : Mission.Current.GetCameraFrame().origin;
+				//Mat3 listenerRotation = GameNetwork.MyPeer.ControlledAgent.LookRotation;
+
+				//float pan = CalculatePan(speakerPosition, listenerPosition, listenerRotation);
+
+				// prevent buffer overflow
+				if (waveProvider.BufferedBytes + byteArray.Length > waveProvider.BufferLength)
+				{
+					waveProvider.ClearBuffer();
+				}
+
+				float pbVolume = 5 * pbOptionVolume;
+
+				// Apply panning to the left and right channels
+				float clampedVolume = CalculateVolume(speakerPosition, listenerPosition, 30f, pbVolume);
+
+				//PBChatHelper.PrintToClientChat(clampedVolume.ToString());
+				//_playerVoiceDataList[k].panProvider.Pan = pan;
+
+				panProvider.Pan = 0.0f;
+				volumeProvider.Volume = clampedVolume;
+
+				if (byteArray.Length <= waveProvider.BufferLength)
+				{
+					waveProvider.AddSamples(byteArray, 0, byteArray.Length);
+				}
+				//this.waveOut.Play();
```

Comment:
Are commented code still useful ?
Yes => Add comment telling why (With TODO maybe?)
No => Remove it

Reply:
Removed